File: Prime-IOS/RootViewController.m
//
//  ViewController.m
//  Prime-IOS
//
//  Created by gypsii001 on 15/4/14.
//  Copyright (c) 2015年 EasyIOS. All rights reserved.
//

#import "RootViewController.h"
#import "PRTabBarController.h"

#import "PRTimeAxisScene.h"
#import "PRMessageScene.h"
#import "PRPersonalScene.h"
#import "PREventScene.h"
#import "PRTabBarIterm.h"
#import "YALAnimatingTabBarConstants.h"
#import "PRNavigationController.h"

@interface RootViewController ()

@property (nonatomic, strong) PRNavigationController *timeNavigationController;
@property (nonatomic, strong) PRNavigationController *newsNavigationController;
@property (nonatomic, strong) PRNavigationController *personNavigationController;
@property (nonatomic, strong) PRNavigationController *activityNavigationController;

@end

@implementation RootViewController
@synthesize tabBarController;

DEF_SINGLETON(RootViewController)

- (void)setupYALTabBarController {
    
    _timeNavigationController = [[PRNavigationController alloc] initWithRootViewController:[[PRTimeAxisScene alloc] init]];

    _newsNavigationController = [[PRNavigationController alloc] initWithRootViewController:[[PRMessageScene alloc]init]];
    
    _activityNavigationController = [[PRNavigationController alloc] initWithRootViewController:[[PREventScene alloc] init]];
    
    _personNavigationController = [[PRNavigationController alloc] initWithRootViewController:[[PRPersonalScene alloc] init]];
    
    
    
    tabBarController = [[PRTabBarController alloc] init];
    
    //prepare leftBarItems
    PRTabBarIterm *item1 = [[PRTabBarIterm alloc] initWithItemImage:[UIImage imageNamed:@"nearby_icon"]
                                                      leftItemImage:[UIImage imageNamed:@"search_icon"]
                                                     rightItemImage:[UIImage imageNamed:@"new_chat_icon"]];
    
    
    PRTabBarIterm *item2 = [[PRTabBarIterm alloc] initWithItemImage:[UIImage imageNamed:@"profile_icon"]
                                                      leftItemImage:nil
                                                     rightItemImage:nil];
    
    tabBarController.leftBarItems = @[item1, item2];
    
    //prepare rightBarItems
    PRTabBarIterm *item3 = [[PRTabBarIterm alloc] initWithItemImage:[UIImage imageNamed:@"chats_icon"]
                                                      leftItemImage:nil
                                                     rightItemImage:nil];
    
    
    PRTabBarIterm *item4 = [[PRTabBarIterm alloc] initWithItemImage:[UIImage imageNamed:@"settings_icon"]
                                                      leftItemImage:[UIImage imageNamed:@"edit_icon"]
                                                     rightItemImage:nil];
    
    tabBarController.rightBarItems = @[item3, item4];
    
    
    tabBarController.centerButtonImage = [UIImage imageNamed:@"plus_icon"];
    
    //customize tabBarView
    tabBarController.tabBarView.extraTabBarItemHeight = YALExtraTabBarItemsDefaultHeight;
    tabBarController.tabBarView.offsetForExtraTabBarItems = YALForExtraTabBarItemsDefaultOffset;
    tabBarController.tabBarView.backgroundColor = [UIColor clearColor];
    tabBarController.tabBarView.tabBarColor = UIColorFromRGB(0x34cdff);
    tabBarController.tabBarViewHeight = YALTabBarViewDefaultHeight;
    tabBarController.tabBarView.tabBarViewEdgeInsets = YALTabBarViewHDefaultEdgeInsets;
    tabBarController.tabBarView.tabBarItemsEdgeInsets = YALTabBarViewItemsDefaultEdgeInsets;
    
    tabBarController.viewControllers = [NSArray arrayWithObjects:_timeNavigationController,_newsNavigationController,_activityNavigationController,_personNavigationController,nil];
    tabBarController.selectedIndex = 0;
    [self.view addSubview:tabBarController.view];
    
}

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    self.view.backgroundColor = [UIColor whiteColor];
    [self setupYALTabBarController];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: Prime-iOS/RootViewController.swift
//
//  ViewController.swift
//  Prime-iOS
//
//  Created by zhu_haibo on 15/8/19.
//  Copyright (c) 2015年 EasyiOS. All rights reserved.
//


import EasyIOS
import FoldingTabBar.YALAnimatingTabBarConstants

class RootViewController: UIViewController {

    var timeNav:PRNavigationController!
    var messageNav:PRNavigationController!
    var personNav:PRNavigationController!
    var eventNav:PRNavigationController!
    
    var tabbarController:PRTabBarController!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        
        setupViewController()
    }

    func setupViewController() {
        timeNav = PRNavigationController(rootViewController: PRTimeAxisScene())
        messageNav = PRNavigationController(rootViewController: PRMessageScene())
        personNav = PRNavigationController(rootViewController: PRPersonalScene())
        eventNav = PRNavigationController(rootViewController: PREventScene())
        
        tabbarController = PRTabBarController()
        
        //prepare leftBarItems
        var item1:PRTabBarIterm = PRTabBarIterm(itemImage:UIImage(named: "nearby_icon"), leftItemImage:UIImage(named: "search_icon"), rightItemImage:UIImage(named: "new_chat_icon"))
        
        var item2:PRTabBarIterm = PRTabBarIterm(itemImage:UIImage(named: "profile_icon"), leftItemImage:UIImage(named: "1"), rightItemImage:UIImage(named: "1"))
        
        tabbarController.leftBarItems = [item1, item2]
        
        
        //prepare rightBarItems
        var item3:PRTabBarIterm = PRTabBarIterm(itemImage:UIImage(named: "chats_icon"), leftItemImage:UIImage(named: "1"), rightItemImage:UIImage(named: "1"))
        
        var item4:PRTabBarIterm = PRTabBarIterm(itemImage:UIImage(named: "settings_icon"), leftItemImage:UIImage(named: "edit_icon"), rightItemImage:UIImage(named: "1"))
        
        tabbarController.rightBarItems = [item3, item4]
        tabbarController.centerButtonImage = UIImage(named: "plus_icon")
        
        
        //customize tabBarView
        tabbarController.tabBarView.extraTabBarItemHeight = YALExtraTabBarItemsDefaultHeight
        
        tabbarController.tabBarView.offsetForExtraTabBarItems = YALForExtraTabBarItemsDefaultOffset;
        tabbarController.tabBarView.backgroundColor = UIColor.clearColor();
        tabbarController.tabBarView.tabBarColor = UIColor(hexString: "#34cdff")
        tabbarController.tabBarViewHeight = YALTabBarViewDefaultHeight;
        tabbarController.tabBarView.tabBarViewEdgeInsets = YALTabBarViewHDefaultEdgeInsets;
        tabbarController.tabBarView.tabBarItemsEdgeInsets = YALTabBarViewItemsDefaultEdgeInsets;
        
        tabbarController.viewControllers = [timeNav, messageNav, eventNav, personNav]
        tabbarController.selectedIndex = 0;
        view.addSubview(tabbarController.view)
        LIVE_LOAD_PATH = __FILE__.stringByDeletingLastPathComponent.stringByAppendingString("/Class/Scene/Xml")  //xml文件的路径（项目路径）
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

